src/main/java/com/carhouse/provider/impl/CarSaleProviderImpl.java

```diff
@@ -1,40 +1,49 @@
 package com.carhouse.provider.impl;
 
-import com.carhouse.model.*;
-import com.carhouse.model.dto.CarDto;
+import com.carhouse.model.CarSale;
 import com.carhouse.model.dto.CarSaleDto;
 import com.carhouse.provider.CarSaleProvider;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.validator.GenericValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
 
-import java.math.BigDecimal;
-import java.sql.Date;
-import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 /**
  * The car sale data provider.
  */
 @Component
 public class CarSaleProviderImpl implements CarSaleProvider {
 
+    @Value("${carSale.url}")
+    private String URL;
+
+    @Value("${car.sale.list.get}")
+    private String CAR_SALE_LIST_GET;
+
+    @Value("${car.sale.get}")
+    private String CAR_SALE_GET;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
     /**
      * Gets list car sale.
      *
+     * @param requestParams the request params
      * @return the list car sale
      */
-    public List<CarSaleDto> getListCarSale() {
-        return new ArrayList<>() {{
-            add(createCarSaleDto(0, "BMW", "M5"));
-            add(createCarSaleDto(1, "Mercedes", "C63AMG"));
-            add(createCarSaleDto(2, "Audi", "RS6"));
-            add(createCarSaleDto(3, "Mercedes", "E220"));
-            add(createCarSaleDto(4, "Mercedes", "C63"));
-            add(createCarSaleDto(5, "Audi", "RS5"));
-            add(createCarSaleDto(6, "Audi", "A8"));
-            add(createCarSaleDto(7, "BMW", "M4"));
-            add(createCarSaleDto(8, "BMW", "8"));
-            add(createCarSaleDto(9, "Mercedes", "GLS"));
-        }};
+    public List<CarSaleDto> getListCarSale(final Map<String, String> requestParams) {
+        CarSaleDto[] listCarSales = restTemplate.getForObject(buildUrl(URL + CAR_SALE_LIST_GET, requestParams),
```

Comment:
1. `listCarSales` contains array. ARRAY!
2. Is it possible to receive ready-to-use list here?

Reply:
Yes, I will be using the exchange method of resttemplate